cmd/frontend/graphqlbackend/org.go

```
@@ -276,7 +276,14 @@ func (r *schemaResolver) RemoveUserFromOrganization(ctx context.Context, args *s
 	if err := database.OrgMembers(r.db).Remove(ctx, orgID, userID); err != nil {
 		return nil, err
 	}
-	r.repoupdaterClient.SchedulePermsSync(ctx, protocol.PermsSyncRequest{UserIDs: []int32{userID}})
+
+	err = r.repoupdaterClient.SchedulePermsSync(ctx, protocol.PermsSyncRequest{UserIDs: []int32{userID}})
+	if err != nil {
+		log15.Warn("schemaResolver.RemoveUserFromOrganization.SchedulePermsSync",
```

qq: do we need to log on the caller side (can't we assume that if there's an error, repoupdaterClient will log)? Or, is log15 (which looks like structured logging) a way to trace a request? 
Not opposed to this change, just asking to understand why we log for a fire-and-forget method (we don't return the error).

---

>qq: do we need to log on the caller side (can't we assume that if there's an error, repoupdaterClient will log)?

I do not think we can, otherwise `SchedulePermsSync` should not even try to return an error.

>just asking to understand why we log for a fire-and-forget method (we don't return the error).

Either way is fine, I think it just depends on the answer of "do we want to block user finished the action if this goes wrong?" In this case, I think we do not need to (but we definitely can depends on our agreement).

---

Apart of what do we need to do with in the case of error, we should not ignore errors that are returned.

---

`log15` does not do any tracing by itself (I'm not expert with it but I think it is possible to inject _something_ to make it do tracing). So far it is just a way to do structured logging.

---

> I do not think we can, otherwise `SchedulePermsSync` should not even try to return an error.
> 
> >

My understanding (of Go) is that SchedulePermsSync will do the same thing whether or not we use the return value - is that correct?

---

> > I do not think we can, otherwise `SchedulePermsSync` should not even try to return an error.
> > >
> 
> My understanding (of Go) is that SchedulePermsSync will do the same thing whether or not we use the return value - is that correct?

Could you spell out what exact do you mean of "will do the same thing"? I was speaking on the function signature design POV, i.e. I the client is going to handle the error, it does/should not propagate further, so either propagate or handle it.

---

I mean that SchedulePermsSync will execute the same code / return the same error whether or not the client uses the value (unlike ex. a "cold" RX.js Observable - not sure if there is a Go equivalent).

Regarding signature / ignoring error - my understanding in general it's fine to keep the error in the signature, since clients can ignore it (ex. if error being returned will not be returned to the user / will not change the control flow). I agree that in this situation we are waiting for the return of SchedulePermsSync, so it makes sense to log the error.

---

> I mean that SchedulePermsSync will execute the same code / return the same error whether or not the client uses the value (unlike ex. a "cold" RX.js Observable - not sure if there is a Go equivalent).

Oh, did not know that with Rxjs, TIL! But in Go, yes, the behavior of call sites has no effect on the function itself.



> Regarding signature / ignoring error - my understanding in general it's fine to keep the error in the signature, since clients can ignore it (ex. if error being returned will not be returned to the user / will not change the control flow). I agree that in this situation we are waiting for the return of SchedulePermsSync, so it makes sense to log the error.

Based on my experience, one of sources of confusion in debugging is ignoring errors that are returned, so in Go, the convention is to always check errors, with exceptions with code comments explaining why ignoring is OK in such cases.

---

Sounds good, thanks!